internals/overlord/state/notices.go

```diff
@@ -81,7 +85,21 @@ type Notice struct {
 }
 
 func (n *Notice) String() string {
-	return fmt.Sprintf("Notice %s (%s:%s)", n.id, n.noticeType, n.key)
+	return fmt.Sprintf("Notice %s (%d:%s:%s)", n.id, n.userID, n.noticeType, n.key)
+}
+
+// UserID returns whether the notice has a user ID, and if so, what the UID is.
+func (n *Notice) UserID() (hasUserID bool, userID uint32) {
```

Comment:
Such functions in Go almost always return the bool second (sometimes called the "comma ok" idiom), including the language's built-in two-variable map access and channel receive. So I suggest we reverse the return value order to `(userID uint32, isSet bool)`. Same with `expandUserID`.